snippets/linear_regression.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+
+def compCostFunction(estim_y, true_y):
+    E = estim_y - true_y
+    C = (1 / 2 * m) * np.sum(E ** 2)
+    return C
+
+def test_dimensions(x, y):
+    # this checks whether the x and y have the same number of samples
+    assert isinstance(x, np.ndarray), "Only works for arrays"
+    assert isinstance(y, np.ndarray), "Only works for arrays"
+    return x.shape[0] == y.shape[0]
+
+# To be deleted later
+# feature_1 = np.linspace(0, 2, num=100)
+
+X = np.random.randn(100,3)  # feature matrix
+y = 1 + np.dot(X, [3.5, 4., -4])  # target vector
+
+m = np.shape(X)[0]  # nr of samples
+n = np.shape(X)[1]  # nr of features
+
+def iterativeLinearRegression(X, y, alpha=0.01):
+    """
+    This makes iterative LR via gradient descent and returns estimated parameters and history list.
+    """
+    steps=500
```

Comment:
This should be an argument to the function rather than hard-coded within it. (please)